src/sbvr-api/odata-response.ts

```diff
@@ -107,11 +109,24 @@ const getFetchProcessingFields = (table: AbstractSqlTable) => {
 		)
 		.map(({ fieldName, dataType }) => {
 			const odataName = sqlNameToODataName(fieldName);
-			return [
-				odataName,
-				(sbvrTypes[dataType as keyof typeof sbvrTypes] as SbvrType)
-					.fetchProcessing,
-			];
+			const sbvrDataType = dataType as keyof typeof sbvrTypes;
+			let fetchProcessingFn = (sbvrTypes[sbvrDataType] as SbvrType)
+				.fetchProcessing;
+
+			// fetchProcessing for WebResources needs to be extended in order
+			// to support presigning a URL immedeately after it is fetched from the database
+			// as no URL can be served without the signing
+			if (sbvrDataType === 'WebResource') {
```

Comment:
@fisehara I tried to solve performance issues by going one step deeper into the stack here. Now instead of having to figure it out if a row/fieldname is a webresource we bind the `fetchProcessing` function incoming from sbvrTypes with the onPreRespond hook. This operation happens before the expensive MxN loop on process and should be fairly cheap, and avoid us having to do any extra checks in there